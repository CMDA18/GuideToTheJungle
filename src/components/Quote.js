// @flow
import React from 'react'
import styled from 'styled-components'
import Quotemark from '../shared/elements/QuoteMark'
import {
  // textStyles,
  colorPalette
} from '../styles/styleGuide'

type Props = {
  image: string,
  text: string,
  metadata: string
}

const Quote = (props: Props) =>
  <Container>
    <Image url={props.image}/>
    <TextContainer>
      <Quotemark />
      <QuoteText>{props.text}</QuoteText>
      <Line />
      <Metadata>{props.metadata}</Metadata>
    </TextContainer>
  </Container>

export default Quote

const Container = styled.div`
  padding: 15px;
  display: flex;
`

const Image = styled.div`
  width: 305px;
  height: 330px;
`

const TextContainer = styled.div`
  background-color: ${colorPalette.lightBlue};
  width: 722px;
`

const QuoteText = styled.p`
`

const Line = styled.span`
  display: block;
  border-bottom: 4px solid ${colorPalette.orange};
  width: 44px;
  height: 5px;
  margin: auto;
`

const Metadata = styled.p`
`
