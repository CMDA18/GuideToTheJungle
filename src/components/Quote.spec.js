import React from 'react'
import { shallow } from 'enzyme'
import Quote from './Quote'

it('renders without crashing', () => {
  expect(shallow(<Quote />)).toMatchSnapshot()
})
