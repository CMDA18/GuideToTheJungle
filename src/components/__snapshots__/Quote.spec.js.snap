// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Quote />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Quote__Image
          backgroundImage={undefined}
        />,
        <Quote__TextContainer>
          <Quote__Wrapper>
            <QuoteMark />
          </Quote__Wrapper>
          <Quote__QuoteText />
          <Quote__Line />
          <Quote__Metadata />
        </Quote__TextContainer>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "backgroundImage": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Quote__Wrapper>
              <QuoteMark />
            </Quote__Wrapper>,
            <Quote__QuoteText />,
            <Quote__Line />,
            <Quote__Metadata />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <QuoteMark />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Quote__Image
            backgroundImage={undefined}
          />,
          <Quote__TextContainer>
            <Quote__Wrapper>
              <QuoteMark />
            </Quote__Wrapper>
            <Quote__QuoteText />
            <Quote__Line />
            <Quote__Metadata />
          </Quote__TextContainer>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backgroundImage": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Quote__Wrapper>
                <QuoteMark />
              </Quote__Wrapper>,
              <Quote__QuoteText />,
              <Quote__Line />,
              <Quote__Metadata />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <QuoteMark />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
